gpu/segmentation/include/pcl/gpu/segmentation/impl/gpu_extract_clusters.hpp

```diff
@@ -41,6 +41,25 @@
 
 #include <pcl/gpu/segmentation/gpu_extract_clusters.h>
 
+#include <cuda_runtime_api.h>
+#include <cuda.h>
+std::vector<int> economical_download(const pcl::gpu::NeighborIndices& indices,
+        std::vector<int>& sizes, int max_answers) {
+    indices.sizes.download(sizes);
+    std::vector<int> results;
+    for (std::size_t qp = 0; qp < sizes.size(); qp++) {
+        int begin = qp * max_answers;
+        const int* pdata = indices.data.ptr() + begin;
+        std::vector<int> tmp(sizes[qp]);
```

Comment:
What do you want to move out of the loop? I think I can move the initialization of the vector out of the loop. I am not so sure about the other two lines.

Reply:
The vector can be moved outside the loop, with the first line of the loop clearing the vector. This will reduce the allocations from the current 1 per iteration.

If needed, we can find the max size and reserve the vector causing exactly 1 allocation

Reply:
Great - then I understood what you mean. Thanks for the explanation. I have moved the construction of the vector outside and the vector is also cleared each time.